libraries/AP_UAVCAN/AP_UAVCAN.cpp

```diff
@@ -322,8 +322,14 @@ void AP_UAVCAN::init(uint8_t driver_index, bool enable_filters)
         return;
     }
 
+    _dna_server = new AP_UAVCAN_DNA_Server(this, StorageAccess(StorageManager::StorageCANDNA));
+    if (_dna_server == nullptr) {
+        debug_uavcan(AP_CANManager::LOG_ERROR, "UAVCAN: couldn't allocate DNA server\n\r");
```

Comment:
It will just not finish initializing the DroneCAN interface, and hence all the devices connected over that will not show up, and should raise the subsequent pre-arms for each of those devices if setup.

This is not unlike any allocations that are done before getting here. We just return for all of those without specifically notifying users. Maybe we should make init be bool and CANManager should notify user of this error.